cdap-client/src/main/java/co/cask/cdap/client/MetaClient.java

```diff
@@ -54,4 +59,23 @@ public Version getVersion() throws IOException, UnauthorizedException {
     HttpResponse response = restClient.execute(HttpMethod.GET, config.resolveURL("version"), config.getAccessToken());
     return ObjectResponse.fromJsonBody(response, Version.class).getResponseObject();
   }
+
+  public Map<String, ConfigEntry> getCDAPConfig() throws IOException, UnauthorizedException {
+    return getConfig("config/cdap");
+  }
+
+  public Map<String, ConfigEntry> getHadoopConfig() throws IOException, UnauthorizedException {
+    return getConfig("config/hadoop");
+  }
+
+  private Map<String, ConfigEntry> getConfig(String url) throws IOException, UnauthorizedException {
+    HttpResponse response = restClient.execute(HttpMethod.GET, config.resolveURL(url), config.getAccessToken());
+    List<ConfigEntry> responseObject =
```

Comment:
Rather than having GSON read through the whole response body and parse it into a `List` then convert to `Map`, I think you can create a custom GSON deserializer. Not a huge deal though.